### PR TITLE
Development/filtering for events

### LINKED ISSUE #325: Rework schedule service
Complete rework for Schedule Service

### PR DESCRIPTION
**Pull request by CptHouston**
Created GetEventOccuranceByID & GetEventsFiltered 
for Schedules controller

-- filtering is done by mentorid/groupid/scheduleid/courseid/eventOccurenceId
-- added accessibility for mentor/admin/student/secretary
-- updated event occurence DTO to include related scheduled events

### CODE REVIEW

File: CharlieBackend.Business/Services/ScheduleServiceFolder/EventHandlers/ScheduledEventHandlerBase.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                EventStart = targetStartDate,
+                EventFinish = targetFinishDate,
+                MentorId = context.MentorID,
+                ThemeId = context.ThemeID
+            };
+        }
+
+        public virtual int GetIterationCount()
+        {
+            return 1;
```
Comment (Hlib-mats) [inline CharlieBackend.Business/Services/ScheduleServiceFolder/EventHandlers/ScheduledEventHandlerBase.cs:54]: what is this magic number?

File: CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+using CharlieBackend.Core;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.DTO.Schedule;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+
```
Comment (Hlib-mats) [inline CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs]: remove pls empty line

File: CharlieBackend.Core/Entities/PatternType.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace CharlieBackend.Core.DTO.Schedule
+{
+    
+}
```
Comment (Hlib-mats) [inline CharlieBackend.Core/Entities/PatternType.cs]: here is nothing

**Review by Hlib-mats — APPROVED** [review]

**Review by andy-dendy-repo — APPROVED** [review]

**Review by Serhii-Dudar — APPROVED** [review]

**Review by filinskyi — APPROVED** [review]

File: CharlieBackend.Business/Services/ScheduleServiceFolder/EventHandlers/ScheduledEventHandlerBase.cs (hunk 1 above)
Comment (CptHouston) [reply CharlieBackend.Business/Services/ScheduleServiceFolder/EventHandlers/ScheduledEventHandlerBase.cs:54]: 1 is a default value for a loop. It is used as is in DayHandler. A method is outlined to enable polymorphism

File: CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs (hunk 2 above)
Comment (CptHouston) [reply CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs]: removed

Comment (CptHouston) [reply CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs]: For basic changes to ScheduleService, please see https://github.com/ita-social-projects/WhatBackend/pull/337
This PR is created for Filtering explicitly

File: CharlieBackend.Core/Entities/PatternType.cs (hunk 3 above)
Comment (CptHouston) [reply CharlieBackend.Core/Entities/PatternType.cs]: file removed

**Comment by CptHouston** [comment]
Should be merged after https://github.com/ita-social-projects/WhatBackend/pull/337

**Review by TetianaDomanska-dev — APPROVED** [review]

**Pull request merged**